code/game/machinery/doors/airlock.dm

```diff
@@ -353,6 +353,8 @@ About the new airlock wires panel:
 		..()
 
 /obj/machinery/door/airlock/bump_open(mob/living/user as mob) //Airlocks now zap you when you 'bump' them open when they're electrified. --NeoFite
+	if(user.loc == loc)	//no bumping an airlock from within the airlock
```

Comment:
Under what circumstances would this happen?

Reply:
When I say "within the airlock" I mean "after being closed inside the airlock."
If an airlock crushes you and you remain on its tile, attempting to move would cause you to bump the airlock as though you were walking into it from outside. I think the person should at least have to use their hands to open it manually after being pinned in the middle of it, assuming they have access.

Reply:
Is `bump_open` really called in that scenario, though? if it is, it would mean people uncrossing the airlock would get shocked, if the airlock is electrified, which is not what happens. I don't think `bump_open` is called at all when uncrossing.

Reply:
I specifically tested it with `to_chat(world)` calls. If I placed myself on an airlock's turf and attempted to move off of it, both the `bump_open()` on line 382 as well as the one on line 355 would be called, and the airlock would open.

Reply:
Also, the addition of this `if()` check inside `bump_open()` successfully stopped the airlock from opening in those circumstances, which further suggests that it is called.

Reply:
well, then

Reply:
@DamianX You `Bump()` something if you *fail* to `Uncross()` it, just like if you fail to `Cross()` it
This is why you can push windows